src/utils/vc/credentials.js

```diff
@@ -316,61 +343,96 @@ export async function verifyCredential(vcJSONorString, {
         controller,
       }),
     // TODO: support more key types, see digitalbazaar github
-    suite: [
-      new Ed25519Signature2018(),
-      new EcdsaSepc256k1Signature2019(),
-      new Sr25519Signature2020(),
-      new Bls12381BBSSignatureDock2022(),
-      new Bls12381BBSSignatureProofDock2022(),
-      new Bls12381BBSSignatureDock2023(),
-      new Bls12381BBSSignatureProofDock2023(),
-      new Bls12381PSSignatureDock2023(),
-      new Bls12381PSSignatureProofDock2023(),
-      new JsonWebSignature2020(),
-      ...suite,
-    ],
+    suite: fullSuite,
     documentLoader: docLoader,
     compactProof,
   });
 
   // Check for revocation only if the credential is verified and revocation check is needed.
-  if (
-    result.verified
-    && isRevocationCheckNeeded(
-      expandedCredential,
-      forceRevocationCheck,
-      revocationApi,
-    )
-  ) {
-    const revResult = await checkRevocationStatus(
-      expandedCredential,
-      revocationApi,
-    );
-
-    // If revocation check fails, return the error else return the result of credential verification to avoid data loss.
-    if (!revResult.verified) {
-      return revResult;
+  if (result.verified && !skipRevocationCheck) {
+    const statuses = getCredentialStatuses(expandedCredential);
```

Comment:
There should never be more than 1 status so `getCredentialStatuses` should return the status if present and be renamed to `getCredentialStatuses`. And we can replace the following switch block